src/utils/lst_gen.rs

```diff
@@ -17,49 +16,70 @@ pub fn update_or_create_status_list(
 
     let bits = bits.value() as usize;
 
-    let mut status_array: Vec<u8> = if let Some(existing_lst) = existing_lst {
-        // Decode & decompress existing list
-        let compressed_data = decode(&existing_lst).map_err(|_| Error::DecodeError)?;
-        let mut decoder = ZlibDecoder::new(&compressed_data[..]);
-        let mut decompressed = Vec::new();
-        decoder
-            .read_to_end(&mut decompressed)
-            .map_err(|e| Error::Generic(e.to_string()))?;
-        decompressed
-    } else {
-        // Determine the highest index to set the size of the status array
-        let max_index = status_updates
-            .iter()
-            .map(|update| update.index)
-            .max()
-            .ok_or_else(|| Error::Generic("Failed to determine max index".to_string()))?;
-
-        if max_index < 0 {
-            return Err(Error::InvalidIndex);
-        }
+    let max_index = status_updates
+        .iter()
+        .map(|update| update.index)
```

Comment:
Directly convert it as usize here will unsure negative numbers are catched earlier.
```suggestion
        .map(|update| update.index as usize)
```

Reply:
This is change is actually risky cause passing a negative index will result to an error when trying to cast it into a usize

Reply:
No, it will not. It will lead to incorrect behavior later in the code. A better solution is to validate the value upfront either by rejecting negative values early or by ensuring only positive values are allowed in the status entry.